include/qpdf/Pl_DCT.hh

```diff
@@ -62,6 +63,10 @@ class QPDF_DLL_CLASS Pl_DCT: public Pipeline
         virtual void apply(jpeg_compress_struct*) = 0;
     };
 
+    QPDF_DLL
+    static std::shared_ptr<CompressConfig>
+        make_compress_config(std::function<void(jpeg_compress_struct*)>);
```

Comment:
I think a unique_ptr would work here. If necessary the caller can let a shared_ptr take ownership.